### PR TITLE
examples/kubernetes: Add tofqdns-enable-poller option

### PR DESCRIPTION
**Pull request by vadorovsky**
Add `tofqdns-enable-poller` option to the Config Map to allow users to
set it on Kubernetes installations.

Fixes: #6870

Signed-off-by: Michal Rostecki <mrostecki@opensuse.org>

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/6949)

### CODE REVIEW

**Comment by vadorovsky** [comment]
@joestringer PTAL

File: examples/kubernetes/templates/v1/cilium-cm.yaml
Comment (vadorovsky) [inline examples/kubernetes/templates/v1/cilium-cm.yaml]: I would like to have a link to the Upgrade Guide here. As soon as we have a short link, I will add it here.

**Review by aanm — APPROVED** [review]

**Comment by aanm** [comment]
waiting on @joestringer approval and @raybejjani

**Review by joestringer — COMMENTED** [review]
I'm still a little bit on the fence as to how easy we should make this for people. We don't really want people to enable this unless they explicitly know they want to.

That said the PR looks good. I agree it'd be good to have a nice `cilium.link/enable-fqdn-poller` link here to point to the relevant section of the upgrade guide.

File: examples/kubernetes/1.10/cilium-cm.yaml
Comment (tgraf) [inline examples/kubernetes/1.10/cilium-cm.yaml]: I would reword this to:
```
# This option is disabled by default starting from version 1.4.x in favor
# of a more powerful DNS proxy-based implementation, see [0] for details.
# Enable this option if you want to use FQDN policies but do not want to use
# the DNS proxy.
# [0] http://docs.cilium.io/en/stable/policy/language/#dns-based
```

The above link will start working as we tag 1.4

**Review by tgraf — CHANGES_REQUESTED** [review]

File: examples/kubernetes/1.10/cilium-cm.yaml
Comment (vadorovsky) [reply examples/kubernetes/1.10/cilium-cm.yaml]: done

File: examples/kubernetes/1.10/cilium-cm.yaml
Comment (vadorovsky) [inline examples/kubernetes/1.10/cilium-cm.yaml:159]: It would be nice to have short links

**Comment by vadorovsky** [comment]
test-me-please

**Comment by tgraf** [comment]
@mrostecki I think adding the environment variable is good. I'm not sure we want to duplicate all of upgrade instructions and information in the ConfigMap. It may be better to not add the example in the default ConfigMap but move what you have as a comment to here instead:
https://docs.cilium.io/en/latest/install/upgrade/#upgrading-from-1-2-5-to-1-4-y

We don't really want users to enable the poller by default.

**Comment by vadorovsky** [comment]
@tgraf There are other options in the ConfigMap which explain the upgrade impact (`t-global-max-entries-tcp`, `ct-global-max-entries-other`, `preallocate-bpf-maps`) and suggest to set different values for easier upgrades, so I just wanted have consistency.

I wrote my comment according to @raybejjani's PR #6886 and the link I put at the end is supposed to redirect to those docs as soon as they are merged and released.

I'm OK with closing this PR if you think that it's unnecessary and that we should rather tell all users to change their policies before upgrade[0]. I just wanted to make clear that my PR is just a tl;dr version of @raybejjani's docs and attempt to have consistency in the CM.

[0] although if we really decide to do that, maybe we should think whether it can be somehow scripted or automated at least a bit?

**Comment by tgraf** [comment]
@mrostecki Good points. @raybejjani is writing the upgrade notes for FQDN. I'm OK to merge this and have a discussion in 1.5 what exactly we want in the ConfigMap and what we want to have in upgrade docs only.

**Comment by tgraf** [comment]
The ConfigMap keys must be fixed up though:
```
 	 The DaemonSet "cilium" is invalid: spec.template.spec.containers[0].env[22].valueFrom.configMapKeyRef.name: Invalid value: "": a DNS-1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')

```

File: examples/kubernetes/1.12/cilium-external-etcd.yaml
Comment (raybejjani) [inline examples/kubernetes/1.12/cilium-external-etcd.yaml:337]: Did we accidentally snip the name & optional lines for CILIUM_MASQUERADE here?

**Review by raybejjani — APPROVED** [review]
We messed up the lines in the configmap for CILIUM_MASQUERADE, I think. Otherwise, we should merge this before https://github.com/cilium/cilium/pull/6886 because it uses the flag we added here :)

File: examples/kubernetes/1.12/cilium-external-etcd.yaml
Comment (vadorovsky) [reply examples/kubernetes/1.12/cilium-external-etcd.yaml:337]: Good catch. I will fix it now.

Comment (vadorovsky) [reply examples/kubernetes/1.12/cilium-external-etcd.yaml:337]: fixed

**Comment by vadorovsky** [comment]
test-me-please

**Comment by vadorovsky** [comment]
test-me-please

**Comment by vadorovsky** [comment]
@tgraf @raybejjani I fixed that. Please look again.

**Review by ianvernon — APPROVED** [review]

**Review by ianvernon — APPROVED** [review]

**Comment by vadorovsky** [comment]
test-me-please

**Review by tgraf — APPROVED** [review]

**Pull request merged**
